File: android_fragment/app/src/main/java/com/support/generator/di/modules/AppModule.java
package com.support.generator.di.modules;

import android.content.Context;

import com.support.generator.di.App;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * @author: leellun
 * @data: 2018/12/20.
 */
@Module
public class AppModule {

  @Singleton
  @Provides
  public Context provideContext(App application) {
    return application.getApplicationContext();
  }
}

File: example/app/src/main/java/gauges/mobile/github/com/mvpapplication/module/di/modules/AppModule.kt
package gauges.mobile.github.com.mvpapplication.module.di.modules

import android.content.Context
import dagger.Module
import dagger.Provides
import gauges.mobile.github.com.mvpapplication.module.di.App
import gauges.mobile.github.com.mvpapplication.module.main.MainActivitySubComponent
import javax.inject.Singleton

/**
 * @author: leellun
 * @data: 2018/12/19.
 *
 */
@Module(subcomponents = arrayOf(
//        register your activities' subcomponents here
        MainActivitySubComponent::class
))
class AppModule {

    @Singleton
    @Provides
    internal fun provideContext(application: App): Context = application.applicationContext
}